### PR TITLE
use github actions to build docker images

### PR DESCRIPTION
**Pull request by KaiSzuttor**
* remove intel and exotic archs
* remove tests with espresso
* removed latex from cuda 10 image

### CODE REVIEW

**Comment by fweik** [comment]
This looks very good to me.

**Comment by RudolfWeeber** [comment]
Cool.

**Comment by KaiSzuttor** [comment]
Do we want to have the scheduled builds? That would mean that the hash-tag is also not really static...

**Comment by RudolfWeeber** [comment]
I think, builds on changes of the Dockerfiles or triggered manually would suffice

**Comment by fweik** [comment]
With this you loos the ability to test PRs. Don't know if that matters.

**Comment by KaiSzuttor** [comment]
> With this you lose the ability to test PRs. Don't know if that matters.

Until there is a workflow file in the master, I have to push to a branch in the repo to trigger the build. However, I trigger the builds twice (for PR and push) then. So if this is merged, I will open a PR where I reenable the trigger for PRs.

File: docker/Dockerfile-clang [hunk 1]
```diff
@@ -49,6 +49,7 @@ RUN useradd -m espresso && usermod -a -G www-data espresso
 USER espresso
 ENV HOME="/home/espresso"
 # install Python3 packages locally
-RUN pip3 install --user requests autopep8==1.3.4 pycodestyle==2.3.1 pylint==2.2.2 astroid==2.1.0 isort==4.3.4 cython==0.26 cmake==3.10.3 cmake-format==0.6.9 pre-commit
 ENV PATH="${PATH}:${HOME}/.local/bin"
+RUN pip3 install --upgrade pip
+RUN pip install --user requests autopep8==1.3.4 pycodestyle==2.3.1 pylint==2.2.2 astroid==2.1.0 isort==4.3.4 cython==0.26 cmake==3.10.3 cmake-format==0.6.9 pre-commit
```
Comment (fweik) [inline docker/Dockerfile-clang:54]: pip3?

Comment (KaiSzuttor) [reply docker/Dockerfile-clang:54]: no, you want to use the updated pip that has now ```3```

**Pull request merged**